### PR TITLE
meta: More In-Depth Contributor Guidelines

### PR DESCRIPTION
**Pull request by lyseah**
This PR addresses the guidelines that were discussed at the open meeting on 3/14/2021.

Changes, requests and discussion are welcome.
This PR will also publish (and freshen up) the current README.md

Additionally, existing templates were replaced as they did not follow GitHubs standard and were outdated on such.
Furthermore, this also adds new templates for Bugs, Feature Requests and Pull Requests.

Please note that "Contributor Guidelines" are actual **Contributor Guidelines**. It very much differentiates between (external) Contributors and (internal) Developers (Collaborator, Members of the Repository). These Guidelines are specifically for external contributors and shall not apply to the core development team present now.

### CODE REVIEW

File: .github/CONTRIBUTOR_GUIDELINES.md
Comment (zontreck) [inline .github/CONTRIBUTOR_GUIDELINES.md:97]: Similarly when fixing an issue, you can also simply say: "Fix: #502"

File: .github/CONTRIBUTOR_GUIDELINES.md
Comment (zontreck) [inline .github/CONTRIBUTOR_GUIDELINES.md:62]: As i was mentioning in the discord. I very much am against any guidelines that do not leave open a path for contribution when there is not anyone to review.

Comment (lyseah) [reply .github/CONTRIBUTOR_GUIDELINES.md:62]: Please note that this guideline very much differentiates between (external) Contributors and Developers (Collaborator, Members of the Repository). These Guidelines are specifically for external contributors and shall not apply to the core development team present now.

File: .github/CONTRIBUTING.md
Comment (zontreck) [inline .github/CONTRIBUTING.md:66]: Correct this to: https://opencollar.cc - See the CNAME file in that repo

File: .github/CONTRIBUTING.md
Comment (zontreck) [inline .github/CONTRIBUTING.md:63]: This does not differentiate between staff and external contributors.  Staff being defined as anyone invited to the github organization with commit rights, and a member of the Staff SL group.  External contributor being defined as someone not with those commit rights.  The guidelines discussed in discord make this a very rigid process that discourages contribution from staff members, due to the requirement of a review for staff contributions.  I agree that if there is more than 1 active staff developer, that there should be a review process, but it should not be in such a way that it would halt development for internal contributions.  During conversations with Nirea, it was made clear that there would be no deadline, that if a PR does not receive a review, it would sit there indefinitely, never being merged in.

I'd like to see something that will encourage contribution and will also protect contributors, as a community driven project, and set a deadline, if not reviewed by X date, if the contribution is from staff, it should be allowed to be merged. The timeline i proposed was 7 days, but this could be changed to 14, for instance, but a deadline is needed or if there is ever only 1 developer, as there is now, then the project comes to an immediate halt.

**Review by zontreck — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md
Comment (Doxylamin) [reply .github/CONTRIBUTING.md:66]: This is referring to the source of the documentation file, and advising contributors to appropriately document them (in a seperate PR in this case), if they propose new features. The domain name is the actual displaying of the contents for the end-user.

Comment (lyseah) [reply .github/CONTRIBUTING.md:66]: This is referring to the source of the documentation file, and advising contributors to appropriately document them (in a seperate PR in this case), if they propose new features. The domain name is the actual displaying of the contents for the end-user.

File: .github/CONTRIBUTING.md
Comment (lyseah) [reply .github/CONTRIBUTING.md:63]: +1 to this. GitHub already does a good job of distinguishing between these two. 

Collaborator: Someone who has been invited to work directly on the project as a **member of the repository**. Has permission to push. Someone who has been actively invited to work on the core development team.

Contributor: someone who contributes something to the project, even if they are not a direct member of the repository.

These guidelines refer to Contributors (hence CONTRIBUTING.md), not Collaborators.

**Comment by lyseah** [comment]
Following policy changes have been discussed and will be further implemented into my PR shortly:

- Developers (Internal) and Collaborators (External) clearly need more communication.
  - Issues and PRs should be discussed and not just left laying around.
- All development shall be done on the development branch and PRs shall only target such.
  - Master will become a freeze-branch for the current stable version available in-world.
  - This ensures that someone who's "fixing" a collar will receive the same files the updater provides them with.
  - Non-trivial changes require an issue with RFC (request for comment), where the implementation and technical details can be discussed
- Code comments may not exceed 5 lines.
  - Only crucial functions shall/must be commented.
  - Comments shall include description of parameters and (expected) result.

**Review by SilkieSabra — COMMENTED** [review]
I would like to edit this document.  Can we open it in a google doc or something more friendly to collaborative editing?

File: .github/CONTRIBUTOR_GUIDELINES.md
Comment (SilkieSabra) [inline .github/CONTRIBUTOR_GUIDELINES.md]: A Table of Contents line here for reviewing pull requests

File: .github/CONTRIBUTOR_GUIDELINES.md
Comment (SilkieSabra) [inline .github/CONTRIBUTOR_GUIDELINES.md]: Replace 3-5 and add here: " This document lists ways you can contribute to developing the OpenCollar source code. Listed here are:  Reporting issues such as bug reports and feature requests, submitting pull requests, and reviewing pull requests.  Other ways you may contribute (not covered here):  Inworld help, asking and answering questions in the group/discord, documentation, checking vendor locations and reporting if they are out of date, beta testing, and building your collars with our scripts.  We welcome any and all of these."

File: .github/CONTRIBUTOR_GUIDELINES.md
Comment (SilkieSabra) [inline .github/CONTRIBUTOR_GUIDELINES.md]: Add a section: ### Reviewing Others' Pull Requests
This is an important way to help OpenCollar, since pull requests from contributors will not be merged without review.

1.  Check the github for outstanding pull requests.
2. Read the pull request
3. Review and either mark approval or suggest further changes.

File: .github/CONTRIBUTOR_GUIDELINES.md
Comment (SilkieSabra) [inline .github/CONTRIBUTOR_GUIDELINES.md]: Table of Contents does not match headings?

File: .github/CONTRIBUTING.md
Comment (nirea) [inline .github/CONTRIBUTING.md:1]: This document overlaps a lot with https://github.com/OpenCollarTeam/opencollarteam.github.io/pull/7.

We need to make sure we don't end up with two conflicting statements of project processes and policies.  I'd like to put the more long form stuff there and keep "CONTRIBUTING.md" to a more barebones summary of how we use Git.  Thoughts?

File: .github/CONTRIBUTING.md
Comment (nirea) [reply .github/CONTRIBUTING.md:63]: There is no "approved by default if not reviewed in X days" rule.

I will not distribute unreviewed code to the inworld system.

File: .github/CONTRIBUTING.md
Comment (lyseah) [reply .github/CONTRIBUTING.md:66]: I'm referring to LSLint in this case. It is being linted on PR, but the contributor should still test/lint it locally before pushing any commits.

File: .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md
Comment (lyseah) [reply .github/ISSUE_TEMPLATE/PULL_REQUEST_TEMPLATE.md:49]: I'm referring to LSLint in this case. It is being linted on PR, but the contributor should still test/lint it locally before pushing any commits.

File: .github/CONTRIBUTING.md
Comment (lyseah) [reply .github/CONTRIBUTING.md:1]: I created these documents as part of the last meeting on 3/14/2021 and was not aware that anyone else was working on such a document. The topic was abruptly ended without making a final decision. Aria and many others (including myself) were in a quandary as to how to handle things. I have received occasional feedback from Aria and Roan during the writing of these guidelines to ensure that they are in line with OpenCollar's goal.

Regarding the seperate repo: My understanding is that all rules/policies for a given repository must be in the same repository. As an on-off contributor, I wouldn't want to check Nx different pages/repos just to figure out how and what to do - and under which policies/guidelines.

**Comment by lyseah** [comment]
Obsolete since https://github.com/OpenCollarTeam/opencollarteam.github.io/pull/7 was introduced / proposed.

**Pull request closed without merging**
